plumbing/transport/ssh/common.go

```diff
@@ -11,7 +11,16 @@ import (
 )
 
 // DefaultClient is the default SSH client.
-var DefaultClient = common.NewClient(&runner{})
+var DefaultClient = NewClient()
+
+// NewClient creates a new SSH client with the given options.
+func NewClient(opts ...ClientOption) transport.Transport {
```

Comment:
You can archive the same just make a copy of the `ssh.ClientConfig`. And the API for the go-git user will be more simple and easy to use.